ocfweb/stats/semester_job.py

```diff
@@ -0,0 +1,109 @@
+from datetime import date
+from datetime import datetime
+from datetime import timedelta
+
+import ocflib.printing.quota as quota
+from django.http import HttpResponse
+from matplotlib.figure import Figure
+from matplotlib.ticker import MaxNLocator
+from ocflib.printing.quota import WEEKDAY_QUOTA
+from ocflib.printing.quota import WEEKEND_QUOTA
+
+from ocfweb.component.graph import canonical_graph
+from ocfweb.component.graph import plot_to_image_bytes
+
+ONE_DAY = timedelta(days=1)
+MONTH_INDEX = 1
+SEMESTER_CUTOFF_MONTH = 8
+
+year = datetime.today().year
+spring_start = date(year, 1, 1)
+fall_start = date(year, 8, 1)
+
+
+def get_start_end():
+    """reutnr a tuple of startdate and endate of the semester based on today"""
+    day = datetime.today().timetuple()
+    if day[MONTH_INDEX] < SEMESTER_CUTOFF_MONTH:
+        return (spring_start, fall_start - ONE_DAY)
+    else:
+        return (fall_start, spring_start - ONE_DAY)
+
+
+@canonical_graph(default_start_end=get_start_end)
+def weekday_jobs_image(request, start_day, end_day):
+    return HttpResponse(
+        plot_to_image_bytes(get_jobs_plot('weekday', start_day, end_day), format='svg'),
+        content_type='image/svg+xml',
+    )
+
+
+@canonical_graph(default_start_end=get_start_end)
+def weekend_jobs_image(request, start_day, end_day):
+    return HttpResponse(
+        plot_to_image_bytes(get_jobs_plot('weekend', start_day, end_day), format='svg'),
+        content_type='image/svg+xml',
+    )
+
+
+# dictionary that gives parameters/config of different job graphs
+graphs = {
+    'weekday':
+    {'query': '''
+            SELECT `pages`, SUM(`count`) AS `count`
+            FROM `public_jobs`
+            WHERE
+                (`pages` <= %s) AND
+                (DAYOFWEEK(`day`) MOD 7 > 1) AND
+                (`day` BETWEEN CAST(%s AS Date) AND CAST(%s AS DATE))
+            GROUP BY `pages`''',
+     'quota': tuple([WEEKDAY_QUOTA]),
+     'title': 'Semester Weekday Jobs Distributiom'},
+    'weekend':
+    {'query': '''
+            SELECT `pages`, SUM(`count`) AS `count`
+            FROM `public_jobs`
+            WHERE
+                (`pages` <= %s) AND
+                (DAYOFWEEK(`day`) MOD 7 <= 1) AND
+                (`day` BETWEEN CAST(%s AS Date) AND CAST(%s AS DATE))
+            GROUP BY `pages`''',
+     'quota': tuple([WEEKEND_QUOTA]),
+     'title': 'Semester Weekend Jobs Distribution'}
+}
```

Comment:
The indentation is confusing here, since it's a dictionary, but it isn't clear it's a dictionary because the key is on a separate line, so I think moving the key and indenting the query more would make this clearer.